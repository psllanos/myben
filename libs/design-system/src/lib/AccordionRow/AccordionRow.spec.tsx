import { fireEvent, render, screen } from '@testing-library/react'
import { AccordionRow } from './'

describe('AccordionRow', () => {
    describe('when rendered with text content', () => {
        it('should display the text', () => {
            const component = render(<AccordionRow label="Label">Hello, World!</AccordionRow>)

            expect(screen.getByText('Label')).toBeInTheDocument()
            expect(screen.getByText('Hello, World!')).toBeInTheDocument()
            expect(component).toMatchSnapshot()
        })
    })

    describe('when passed an `href` prop', () => {
        it('should render as a link', () => {
            const component = render(
                <AccordionRow label="Label" href="/to">
                    Hello, World!
                </AccordionRow>
            )

            expect(screen.getByRole('link')).toHaveAttribute('href', '/to')
            expect(component).toMatchSnapshot()
        })
    })

    describe('when passed an `active` prop', () => {
        it('should be highlighted', () => {
            const component = render(
                <AccordionRow label="Label" active={true}>
                    Hello, World!
                </AccordionRow>
            )

            expect(component).toMatchSnapshot()
        })
    })

    describe('when passed a `collapsible` prop', () => {
        it('should render a caret for expanding and collapsing when collapsible', () => {
            const component = render(
                <AccordionRow label="Label" collapsible={true}>
                    Hello, World!
                </AccordionRow>
            )

            expect(component).toMatchSnapshot()
        })

        it('should not render a caret when not collapsible', () => {
            const component = render(
                <AccordionRow label="Label" collapsible={false}>
                    Hello, World!
                </AccordionRow>
            )

            expect(component).toMatchSnapshot()
        })
    })

    describe('when pressed', () => {
        it('should call the `onClick` callback', () => {
            const onClickMock = jest.fn()
            render(<AccordionRow onClick={onClickMock} collapsible={true} label="Label" />)

            fireEvent.click(screen.getByRole('button'))
            expect(onClickMock).toHaveBeenCalledTimes(1)
        })

        it('should call the `onToggle` callback if collapsible', () => {
            const onToggleMock = jest.fn()
            render(<AccordionRow onToggle={onToggleMock} collapsible={true} label="Label" />)

            fireEvent.click(screen.getByRole('button'))
            expect(onToggleMock).toHaveBeenCalledWith(false)

            fireEvent.click(screen.getByRole('button'))
            expect(onToggleMock).toHaveBeenCalledWith(true)
        })
    })
})
