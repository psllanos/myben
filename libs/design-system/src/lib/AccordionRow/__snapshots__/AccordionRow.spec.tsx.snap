// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionRow when passed a \`collapsible\` prop should not render a caret when not collapsible 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-3 pr-3 mb-0.5 flex items-center justify-between space-x-3 rounded-lg leading-none text-base text-white pl-3 bg-gray-500"
      >
        <div
          class="w-full"
        >
          Label
        </div>
      </div>
      <div
        aria-hidden="false"
        class="rah-static rah-static--height-auto "
        style="height: auto; overflow: visible;"
      >
        <div>
          Hello, World!
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-3 pr-3 mb-0.5 flex items-center justify-between space-x-3 rounded-lg leading-none text-base text-white pl-3 bg-gray-500"
    >
      <div
        class="w-full"
      >
        Label
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto "
      style="height: auto; overflow: visible;"
    >
      <div>
        Hello, World!
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AccordionRow when passed a \`collapsible\` prop should render a caret for expanding and collapsing when collapsible 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-3 pr-3 mb-0.5 flex items-center justify-between space-x-3 rounded-lg leading-none text-base text-white select-none pl-3 bg-gray-500 hover:bg-gray-400 cursor-pointer"
        role="button"
      >
        <div
          class="shrink-0 w-4 text-gray-50"
        >
          <svg
            class="transform rotate-90 w-5 h-5 transition-transform"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16 12l-6 6V6z"
              />
            </g>
          </svg>
        </div>
        <div
          class="w-full"
        >
          Label
        </div>
      </div>
      <div
        aria-hidden="false"
        class="rah-static rah-static--height-auto "
        style="height: auto; overflow: visible;"
      >
        <div>
          Hello, World!
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-3 pr-3 mb-0.5 flex items-center justify-between space-x-3 rounded-lg leading-none text-base text-white select-none pl-3 bg-gray-500 hover:bg-gray-400 cursor-pointer"
      role="button"
    >
      <div
        class="shrink-0 w-4 text-gray-50"
      >
        <svg
          class="transform rotate-90 w-5 h-5 transition-transform"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16 12l-6 6V6z"
            />
          </g>
        </svg>
      </div>
      <div
        class="w-full"
      >
        Label
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto "
      style="height: auto; overflow: visible;"
    >
      <div>
        Hello, World!
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AccordionRow when passed an \`active\` prop should be highlighted 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-3 pr-3 mb-0.5 flex items-center justify-between space-x-3 rounded-lg leading-none text-base text-white select-none !bg-cyan !bg-opacity-10 pl-3 bg-gray-500 hover:bg-gray-400 cursor-pointer"
        role="button"
      >
        <div
          class="shrink-0 w-4 text-gray-50"
        >
          <svg
            class="transform rotate-90 w-5 h-5 transition-transform"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16 12l-6 6V6z"
              />
            </g>
          </svg>
        </div>
        <div
          class="w-full"
        >
          Label
        </div>
      </div>
      <div
        aria-hidden="false"
        class="rah-static rah-static--height-auto "
        style="height: auto; overflow: visible;"
      >
        <div>
          Hello, World!
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-3 pr-3 mb-0.5 flex items-center justify-between space-x-3 rounded-lg leading-none text-base text-white select-none !bg-cyan !bg-opacity-10 pl-3 bg-gray-500 hover:bg-gray-400 cursor-pointer"
      role="button"
    >
      <div
        class="shrink-0 w-4 text-gray-50"
      >
        <svg
          class="transform rotate-90 w-5 h-5 transition-transform"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16 12l-6 6V6z"
            />
          </g>
        </svg>
      </div>
      <div
        class="w-full"
      >
        Label
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto "
      style="height: auto; overflow: visible;"
    >
      <div>
        Hello, World!
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AccordionRow when passed an \`href\` prop should render as a link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="/to"
      >
        <div
          class="py-3 pr-3 mb-0.5 flex items-center justify-between space-x-3 rounded-lg leading-none text-base text-white select-none pl-3 bg-gray-500 hover:bg-gray-400 cursor-pointer"
          role="button"
        >
          <div
            class="shrink-0 w-4 text-gray-50"
          >
            <svg
              class="transform rotate-90 w-5 h-5 transition-transform"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M16 12l-6 6V6z"
                />
              </g>
            </svg>
          </div>
          <div
            class="w-full"
          >
            Label
          </div>
        </div>
        <div
          aria-hidden="false"
          class="rah-static rah-static--height-auto "
          style="height: auto; overflow: visible;"
        >
          <div>
            Hello, World!
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="/to"
    >
      <div
        class="py-3 pr-3 mb-0.5 flex items-center justify-between space-x-3 rounded-lg leading-none text-base text-white select-none pl-3 bg-gray-500 hover:bg-gray-400 cursor-pointer"
        role="button"
      >
        <div
          class="shrink-0 w-4 text-gray-50"
        >
          <svg
            class="transform rotate-90 w-5 h-5 transition-transform"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16 12l-6 6V6z"
              />
            </g>
          </svg>
        </div>
        <div
          class="w-full"
        >
          Label
        </div>
      </div>
      <div
        aria-hidden="false"
        class="rah-static rah-static--height-auto "
        style="height: auto; overflow: visible;"
      >
        <div>
          Hello, World!
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AccordionRow when rendered with text content should display the text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="py-3 pr-3 mb-0.5 flex items-center justify-between space-x-3 rounded-lg leading-none text-base text-white select-none pl-3 bg-gray-500 hover:bg-gray-400 cursor-pointer"
        role="button"
      >
        <div
          class="shrink-0 w-4 text-gray-50"
        >
          <svg
            class="transform rotate-90 w-5 h-5 transition-transform"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16 12l-6 6V6z"
              />
            </g>
          </svg>
        </div>
        <div
          class="w-full"
        >
          Label
        </div>
      </div>
      <div
        aria-hidden="false"
        class="rah-static rah-static--height-auto "
        style="height: auto; overflow: visible;"
      >
        <div>
          Hello, World!
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="py-3 pr-3 mb-0.5 flex items-center justify-between space-x-3 rounded-lg leading-none text-base text-white select-none pl-3 bg-gray-500 hover:bg-gray-400 cursor-pointer"
      role="button"
    >
      <div
        class="shrink-0 w-4 text-gray-50"
      >
        <svg
          class="transform rotate-90 w-5 h-5 transition-transform"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16 12l-6 6V6z"
            />
          </g>
        </svg>
      </div>
      <div
        class="w-full"
      >
        Label
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto "
      style="height: auto; overflow: visible;"
    >
      <div>
        Hello, World!
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
